test_rclcpp/test/test_timer.cpp

```diff
@@ -26,8 +26,20 @@
 # define CLASSNAME(NAME, SUFFIX) NAME
 #endif
 
-TEST(CLASSNAME(test_time, RMW_IMPLEMENTATION), timer_fire_regularly) {
-  if (!rclcpp::ok()) {rclcpp::init(0, nullptr);}
+class CLASSNAME (test_time, RMW_IMPLEMENTATION) : public ::testing::Test {
+public:
+  void SetUp() {
+    rclcpp::init(0, nullptr);
+  }
+
+  void TearDown() {
+    if (rclcpp::ok()) {
```

Comment:
Same question as above; do we really need to check `rclcpp::ok` here?

Reply:
In this test, `timer_during_wait` calls shutdown to stop a `spin()`, but `executor.cancel()` can be used instead. Switched.